src/main/scala/dpla/ingestion3/EnricherMain.scala

```diff
@@ -0,0 +1,124 @@
+
+/**
+ * Driver for running an enrichment
+ *
+ * This is a prototype that is hardcoded to run the spatial enrichment,
+ * SpatialEnricher.
+ *
+ * See https://digitalpubliclibraryofamerica.atlassian.net/wiki/display/TECH/Ingestion+3+Storage+Specification#Ingestion3StorageSpecification-S3BucketFolderStructureandContents
+ * with suggestions for how to organize input and output locations. 
+ *
+ * Example invocation:
+ * $ sbt "run-main dpla.ingestion3.EnricherMain local[2] <from> <to>"
+ * ... where <from> is the Avro resource (file / directory) and <to> is the
+ *     destination (directory or s3 folder)
+ *
+ */
+object EnricherMain {
+
+  // Avro schema for our output
+  val schemaStr: String =
+    """
+    {
+      "namespace": "la.dp.avro",
+      "type": "record",
+      "name": "MAPRecord.v1",
+      "doc": "Prototype enriched record",
+      "fields": [
+        {"name": "id", "type": "string"},
+        {"name": "document", "type": "string"}
+      ]
+    }
+    """
+  val logger = LogManager.getLogger(EnricherMain.getClass)
+
+  /*
+   * args:
+   *   master: String -- Spark master descriptor
+   *   inputURI: String -- URI of Avro input
+   *   outputURI: String -- URI of Avro output
+   */
+  def main(args: Array[String]) {
+
+    if (args.length != 3) {
+      logger.error("Arguments should be <master>, <inputURI>, <outputURI>")
+      sys.exit(-1)
+    }
+    val sparkMaster = args(0)
+    val inputURI = args(1)
+    val outputURI = args(2)
+
+    val start_time = System.currentTimeMillis()
+
+    val sparkConf = new SparkConf()
+      .setAppName("Enricher")
+      .setMaster(sparkMaster)
+    val spark = SparkSession.builder().config(sparkConf).getOrCreate()
+    val sc = spark.sparkContext
+
+    val rdd: RDD[Row] = spark.read
+      .format("com.databricks.spark.avro")
+      .option("avroSchema", schemaStr)
+      .load(inputURI)
+      .rdd
+
+    // Process all of the records and wrap them with a dataframe
+    val rows = rdd.map(
+      record => (
+        record.getAs[String]("id"),
+        process(record.getAs[String]("document"))
+      )
+    )
+    val dataframe = spark.createDataFrame(rows)
+
+    val recordCount = dataframe.count()
+    Utils.deleteRecursively(new File(outputURI))
+    dataframe.write
+             .format("com.databricks.spark.avro")
+             .option("avroSchema", schemaStr)
+             .avro(outputURI)
+    sc.stop()
+
+    val end_time = System.currentTimeMillis()
+    Utils.printResults((end_time - start_time), recordCount)
+
+  }
+
+  def process(rdfString: String): String = {
+    try {
+      val rdfByteStream = new StringReader(rdfString)
+      val in_model = Rio.parse(rdfByteStream, "", RDFFormat.TURTLE)
+      val out_model: Model = Spatial.enrich(in_model)
+      // out_model could be passed to other enrichments in turn
+      val out = new StringWriter()
+      Rio.write(out_model, out, RDFFormat.TURTLE)
+      out.toString
+    } catch {
+      case e: Exception => {
```

Comment:
Ignore for now considering my previous comment about wrappers happening at a higher level.